### PR TITLE
Update ESLint

### PR DESCRIPTION
**Pull request by AbdullahAli**
A few updates to make the code nicer

- [x] Use the most up-to-date ESLint and 2018 syntax (async/await, spread, etc)
- [x] Force object spread `...` (🙌🏽) over `Object.assign` (🤢)

### CODE REVIEW

File: app/services/queue.js [hunk 1]
```diff
@@ -504,39 +503,38 @@ class QueueApp extends EventEmitter {
...
-					for (const err of parsedErrorObj.errors) {
-						// Checking for 1902 generation rejection/esp suppression
-						if (err.code !== '1902') {
-							logger('error', {
-								event: 'ON_SINGLE_SEND_TASK_FAILED',
-								err
-							});
-							sendToSentry(email);
+		}).then(logRejections)
+			.catch((errorObj) => {
```
Comment (AbdullahAli) [inline app/services/queue.js]: Not sure why github makes it look like that (it isn't)

File: app/controllers/send.controller.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ const scsCategoryMap = new Map([
 function selectProductInfo(domain, userInfo) {
 	const categorization = userInfo.categorization.find(c => c.domain === domain);
 	//TODO: get rid of lodash
-	return Object.assign({ product: userInfo.product }, _.pick(categorization, 'binding', 'returnPath', 'category'));
+	return { product: userInfo.product, ..._.pick(categorization, 'binding', 'returnPath', 'category') };
```
Comment (keirog) [inline app/controllers/send.controller.js]: Yessss

File: app/util/partially-apply-logger.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 module.exports = ({ rawLogger, data }) => {
 	return (logLevel, extraData) => {
-		rawLogger[logLevel](Object.assign({}, data, extraData));
+		rawLogger[logLevel]({ ...data, ... extraData });
```
Comment (keirog) [inline app/util/partially-apply-logger.js]: ~`... extraData`~ `...extraData`

File: test/middleware/emailContent.test.js [hunk 4]
```diff
@@ -57,7 +57,7 @@ describe('EmailContent middleware', () => {
 
 	it('throws error if fetched email content is marked inactive', done => {
 		nextSpy = sandbox.spy();
-		const inactiveEmailContent = Object.assign({}, emailContent, { active: false });
+		const inactiveEmailContent = { ... emailContent, active: false };
```
Comment (keirog) [inline test/middleware/emailContent.test.js]: ~`... emailContent`~ `...emailContent`

File: test/models/userModel.test.js [hunk 5]
```diff
@@ -34,7 +34,7 @@ describe('User Model', () => {
 	});
 
 	it('updates an existing user', function* () {
-		const updatedDetails = Object.assign({}, userData, { description: 'updated user' });
+		const updatedDetails = { ... userData, description: 'updated user' };
```
Comment (keirog) [inline test/models/userModel.test.js]: ~`... userData~ `...userData`

File: test/util/substitutionData.test.js [hunk 6]
```diff
@@ -10,14 +10,14 @@ const userSubData = [{
 
 describe('The substitutionData util', () => {
 	describe('merge', () => {
-		const recipients = [Object.assign({}, userInfo)];
+		const recipients = [{... userInfo }];
```
Comment (keirog) [inline test/util/substitutionData.test.js]: ~`... userInfo`~ `...userInfo`

File: test/util/substitutionData.test.js [hunk 7]
```diff
@@ -10,14 +10,14 @@ const userSubData = [{
...
 		it('adds user data as substitution data', () => {
 			merge(recipients, userSubData);
 			expect(recipients[0]).to.contain.any.keys(Object.keys(subData));
 		});
 
 		it('does not do anything if no user substitution data', () => {
-			const recipients = [Object.assign({}, userInfo)];
+			const recipients = [{... userInfo }];
```
Comment (keirog) [inline test/util/substitutionData.test.js]: ~`... userInfo`~ `...userInfo`

**Comment by AbdullahAli** [comment]
Ah, good spot @keirog 

I have added eslint rules to ensure these don't get missed next time (re: `...` spacing)

Fixed: https://github.com/Financial-Times/email-send-api/pull/262/commits/f790167b18dba86d93eb43a95dd8f8fba8580b1a

**Comment by AbdullahAli** [comment]
It would be good to extend established eslint styleguides - like [Airbnb Styleguide](https://github.com/airbnb/javascript)

**Pull request merged**